app/models/application_choice.rb

```diff
@@ -9,4 +10,10 @@ class ApplicationChoice < ApplicationRecord
     enrolled: 'enrolled',
     rejected: 'rejected',
   }
+
+private
+
+  def set_id
+    self.id = SecureRandom.hex(5)
```

Comment:
I'm not familiar with how to generate unique IDs that don't clash, but I assume this is sensible?

I've done a bit of reading https://softwareengineering.stackexchange.com/a/130298 and it seems like the chance of a collision should be astronomically low.

Reply:
AFAIK the traditional way is to have a `do` loop like this

```
do
    token = generate_token
    break if token_is_unique_in_the_database(token)
end

# token is now unique
```

Reply:
Updated in: b71a11b72889fa2a1ff86c0c5ee1baae83220e6c